cmd/db.go

```diff
@@ -0,0 +1,95 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	db "github.com/cosmos/cosmos-db"
+)
+
+const (
+	rocksDBMagic  = "\xf7\xcf\xf4\x85\xb7\x41\xe2\x88"
+	pebbleDBMagic = "\xf0\x9f\xaa\xb3\xf0\x9f\xaa\xb3" // 🪳🪳
+	levelDBMagic  = "\x57\xfb\x80\x8b\x24\x75\x47\xdb"
+	magicLen      = 8
+)
+
+func GetFormat(path string) (db.BackendType, error) {
+	magic, err := getMagic(path)
+	if err != nil {
+		return "invalid", err
+	}
+	switch string(magic) {
+	case rocksDBMagic:
+		// Bera returns rocksdb but it's pebble
+		return db.PebbleDBBackend, nil
+	case pebbleDBMagic:
+		return db.PebbleDBBackend, nil
+	case levelDBMagic:
+		return db.GoLevelDBBackend, nil
+
+	}
+	return "invalid", fmt.Errorf("db type not supported, magic: %v", magic)
+}
+
+// Find a file with .sst or .ldb extension
+// read magicLen bytes from the end, return them
+func getMagic(path string) ([]byte, error) {
```

Comment:
1/5 considering this fn operates on a directory, and the `readMagic` operates on the specific file, maybe it would just

```suggestion
func getMagic(pathPrefix string) ([]byte, error) {
```

or

```suggestion
func getMagic(dirPath string) ([]byte, error) {
```

or maybe even that may be something generic, taking `[".sst", ".ldb]` as an argument, but i feel like it's bike shedding at this point

Reply:
i've got a few PRs stacked on top of this one, i'll make a note of this and rename it after merging them all